app/Http/Controllers/MessagesController.php

```diff
@@ -2,20 +2,37 @@
 
 namespace Gladiator\Http\Controllers;
 
+use Gladiator\Models\Contest;
 use Gladiator\Models\Message;
 use Gladiator\Events\QueueMessageRequest;
 
 class MessagesController extends Controller
 {
     /**
+     * Display the specified resource.
+     *
+     * @param  string  $id  Contest ID
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $messages = Message::where('contest_id', '=', $id)->get();
+
+        return view('messages.show', compact('messages'));
+    }
+
+    /**
      * Fire an event that queues a message to be sent.
      *
      * @param int $id
      */
-    public static function sendMessage($id)
+    public static function sendMessage(Message $message)
```

Comment:
There were some issues with that but @angaither updated the route to be 

` Route::get('messages/{message}/send', 'MessagesController@sendMessage')->name('messages.send');`

and some laravel magic made it work. It seems that passing the "message" in to the route definition works like some sorta type-hinting .